### DISCUSSION TITLE
OpenSSL C_Verify*

### DISCUSSION THREAD

**Discussion by 0pq76r** [openssl-projects/pkcs11-provider#321] [Q&A] [answered]
Is it possible to force OpenSSL to use the C_Verify* methods instead of exporting the public key and run the verify locally? Looking at the logs after running `make check`, I get the impression that OpenSSL always prefers the latter.

**Comment by simo5** [comment, marked as answer]
Yes, you should be able to do that via a quirk that prohibits exporting even public keys:

pkcs11-module-allow-export = 1

see https://github.com/latchset/pkcs11-provider/blob/main/docs/provider-pkcs11.7.md#pkcs11-module-allow-export

**Reply by 0pq76r** [reply]
Thanks! I tested it with softhsm and it works as expected.
